### PR TITLE
Fixed: Object likes

### PR DESCRIPTION
**Pull request by jovannypcg**
**Pull request fixed!**

I wrote a function to get the total number of people who liked an object.
An "object" stands for: post, comment, link, status update, photo and so on.

This function is: objectLikes/2

The object is searched by its id and then the number of likes is retrieved.

### CODE REVIEW

File: lib/facebook.ex [hunk 1]
```diff
@@ -213,6 +213,37 @@ defmodule Facebook do
...
+  Gets the total number of people who liked an object.
+  An *object* stands for: post, comment, link, status update, photo.
+
+  If you want to get the likes of a page, please see *pageLikes*.
+
+  ## Example
+      iex> Facebook.objectLikes("1326382730725053_1326476257382367", "<Token>")
+      2
+
+  See: developers.facebook.com/docs/graph-api/reference/object/likes
```
Comment (mweibel) [inline lib/facebook.ex]: `https://` missing

**Comment by mweibel** [comment]
Thanks for your contribution!

When I tried doing that call it never had a summary and also no total_count. Might be that I did something wrong though ;) could you show me an example (e.g. graph API explorer) where this succeeds?

**Comment by jovannypcg** [comment]
Hi @mweibel,

This is an example:

I'd like to get the likes of the post below using the Graph API:

github_fb

To get the ID of that post, I use the Graph API as follows:

post_id

After getting the post ID, a GET request is made in order to retrieve the number of likes:

likes_summary

This function is very useful when you want to analyze a feed of posts one by one.
In addition, I am also working on the function to retrieve the comments. It is almost the same as the objectLikes.

Thanks for your feedback!

**Comment by mweibel** [comment]
Ok, I understand. Probably the tries I did were wrong but could reproduce your way now. 

Would it make sense to be able to retrieve the likes itself instead of just the count? (or alternatively name the function a little different)?

Thanks!

**Comment by jovannypcg** [comment]
Here is the same example, but using facebook.ex:

fbex

**Comment by jovannypcg** [comment]
There is another way to retrieve the likes:

`GET /262588213843476_801732539929038?fields=likes`

But this has a limitation, and it is that facebook uses pagination to get all the likes. If you run that request, you will only get 25 likes of 47. That's why I thought `/likes?summary=true` was the best choise.

What do you recommend to enhance this funciton? :)

**Comment by mweibel** [comment]
`objectLikesSummary`? which calls `objectLikes` with param `:summary: true` and then does the json _parsing_ in it's function?

**Comment by jovannypcg** [comment]
Oh, you're right!
I am going to work on it, just give me a few hours and show you my proposal :)

**Comment by mweibel** [comment]
Great! Might be that i can look at it only tomorrow but let's see ;-)

**Comment by jovannypcg** [comment]
@mweibel,

What do you think about the last commit? :)

File: lib/facebook.ex [hunk 2]
```diff
@@ -213,6 +213,77 @@ defmodule Facebook do
...
+
+  """
+  Provides the summary of a GET request when the 'summary' query parameter is
+  set to true.
+
+  ## Example
+      iex> _objectSummary("comments", "1326382730725053_1326476257382367", "<Token>")
+      %{"total_count" => 47}
+  """
+  defp _objectSummary(scope, object_id, access_token) do
```
Comment (mweibel) [inline lib/facebook.ex]: please remove the `_` in front of the function name. The function is anyway private due to `defp` :)

**Comment by mweibel** [comment]
Looks great! one small comment but otherwise great :)

**Comment by jovannypcg** [comment]
Done, thanks for your feedback :smile:

File: lib/facebook.ex [hunk 3]
```diff
@@ -213,6 +213,77 @@ defmodule Facebook do
...
+  ## Example
+      iex> Facebook.objectCount(:likes, "1326382730725053_1326476257382367", "<Token>")
+      2
+
+  See: https://developers.facebook.com/docs/graph-api/reference/object/likes
+  """
+  def objectCount(:likes, object_id, access_token) do
+    :likes
+      |> Atom.to_string
+      |> objectSummary(object_id, access_token)
```
Comment (mweibel) [inline lib/facebook.ex]: actually, now that I think about it.. Why do you pass :likes as an atom when you convert it to string anyway? 
Couldn't you just do:
```
objectSummary("likes", object_id, access_token)
  |> summaryCount
```

**Comment by jovannypcg** [comment]
I thought it was a better practice to use atoms instead of strings.
Do you recommend to use strings for this function?

**Comment by mweibel** [comment]
You could still use an atom as the argument to the function but then use a string inside the function, no? Because the actual argument is not assigned to a variable which you'd then use to call objectSummary

**Comment by jovannypcg** [comment]
Oh, I get the point, hehe... done :)

**Pull request merged**